src/components/schedule/schedule-time-discover/ScheduleTimeSlot.tsx

```diff
@@ -57,6 +75,92 @@ const ScheduleTimeSlot: FC<ScheduleTimeSlotProps> = ({
     : false
 
   const { isTopElement, isBottomElement } = getMeetingBoundaries(slot, duration)
+
+  // Find event details for current user if they're unavailable
+  const currentUserEvent = useMemo(() => {
```

Comment:
Move this into the hook that generates the slotData that already handles all forms of interpolation from slot intersections